### PR TITLE
Add updater to linux build scripts

### PR DESCRIPTION
**Pull request by gabriel**
Builds updater to be installed and run at /usr/bin/kbupdater

Also fixes kbfs bin name differences for linux.

@oconnor663 

cc @keybase/updater-hackers 

This assumes go-updater is checked out in the go path? Thats not a correct assumption tho?

Logs all starts to `keybase.start.log`

### CODE REVIEW

File: packaging/linux/run_keybase [hunk 1]
```diff
@@ -20,6 +20,9 @@ fi
 if fusermount -uz /keybase &> /dev/null ; then
   echo Unmounting /keybase...
 fi
+if killall kbupdater &> /dev/null ; then
+  echo Shutting down updater...
+fi
```
Comment (oconnor663) [inline packaging/linux/run_keybase]: I'd like to keep the `fusermount -u` and the `killall kbfsfuse` commands next to each other. Maybe move this down one?

File: packaging/linux/run_keybase [hunk 2]
```diff
@@ -42,12 +45,13 @@ logdir="${XDG_CACHE_HOME:-$HOME/.cache}/keybase"
 mkdir -p "$logdir"
 
 echo Launching keybase service...
-keybase service &>> "$logdir/keybase.service.log" &
+keybase -d --log-file="$logdir/keybase.service.log" service &>> "$logdir/keybase.start.log" &
```
Comment (oconnor663) [inline packaging/linux/run_keybase]: Has `keybase log send` been updated to reflect what's happening here?

Comment (oconnor663) [inline packaging/linux/run_keybase]: I wonder if the logging-to-two-different-files approach will work well when/if we switch to a systemd way of doing things. How does it work with launchd on OSX?

File: packaging/linux/build_binaries.sh [hunk 3]
```diff
@@ -5,6 +5,7 @@ set -e -u -o pipefail
 here="$(dirname "$BASH_SOURCE")"
 this_repo="$(git -C "$here" rev-parse --show-toplevel)"
 kbfs_repo="$(dirname "$this_repo")/kbfs"
+updater_repo="$(dirname "$this_repo")/go-updater"
```
Comment (oconnor663) [inline packaging/linux/build_binaries.sh]: I didn't include this in CI, but now we probably should. That's done in the script that calls this one, `build_and_push_packages.sh`, using your `wait-ci` command.

Comment (oconnor663) [inline packaging/linux/build_binaries.sh]: We'll also need to include getting this repo and exposing it to the docker container in `docker_build.sh`.

**Comment by gabriel** [comment]
Jira issue re: custom linux messaging for os notifications from updater: https://keybase.atlassian.net/browse/DESKTOP-1204

**Comment by oconnor663** [comment]
@gabriel I pushed some extra stuff for the Linux packaging. Seems to make the build work on my box.

I haven't tested actually _running_ the updater (or `run_keybase.sh`) with one of these packages, so we'll want to do that after we figure out the story with server-generated-please-update-your-machine messages.

**Comment by gabriel** [comment]
🚢

**Comment by oconnor663** [comment]
My commits are in this PR now. I don't want to land this whole thing until we've gotten a chance to test it on Linux, though. Can we do that Monday?

**Comment by gabriel** [comment]
sure

**Comment by oconnor663** [comment]
Talking to @cjb, I have some worries about what this updater is going to do on Linux. It seems likely that we'll be moving everyone to a daily updates channel (as opposed to what we do with our prod users today, where they only get updates when do a real version bump). In that world, I think every Keybase Linux user would see our update toast every day, regardless of their actual update behavior, and with no obvious way to turn it off. (As opposed to the OSX experience, where you can just check a box and never see it again.) That could be annoying. It's also less valuable, because there's no way for the notification to actually help you perform the update. All it is is a reminder.

I think what we need on Linux is going to be something that notifies you only if you're "several days ish" out of date. If we wanted to put logic like that into the updater, then it would make sense to run the updater on Linux. But if that's more complicated / Linux-specific than the updater wants to be, then I think we're going to want a different mechanism on Linux, and I don't know if the updater will serve a purpose at that point.

**Comment by gabriel** [comment]
Sure... we could re-enable linux builds without the updater and rely on the system package manager to notify the user for a bit. Or we could change the snooze time on linux to something like 3 days (or longer) until we do something a little smarter? AFAIK the snooze time can be variable at each snooze request.

**Comment by oconnor663** [comment]
Just put up https://github.com/keybase/client/pull/3092 with the non-updater bits from this PR.

**Pull request closed without merging**